### PR TITLE
template: add pull request template

### PR DESCRIPTION
**Pull request by lilin90**
@CaitinChen PTAL
Via: https://github.com/pingcap/tidb/pull/6718 & https://github.com/pingcap/tikv/pull/3154

### CODE REVIEW

File: pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+- New feature (non-breaking change which adds functionality)
+- Improvement (non-breaking change which is an improvement to an existing feature)
+- Bug fix (non-breaking change which fixes an issue)
+- Breaking change (fix or feature that would cause existing functionality to not work as expected)
+
+## How has this PR been tested? (mandatory)
+
+Please describe the tests that you ran to verify your changes. Have you finished unit tests, integration tests, or manual tests?
+
+## Does this PR affect documentation (docs/docs-cn) update? (optional)
```
Comment (queenypingcap) [inline pull_request_template.md]: This would be unnecessary because this is the document repo...

File: pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+If there is document change, please file a PR in ([docs](https://github.com/pingcap/docs) and [docs-cn](https://github.com/pingcap/docs-cn)) and add the PR number here.
+
+## Refer to a related PR or issue link (optional)
+
+## Benchmark result if necessary (optional)
```
Comment (queenypingcap) [inline pull_request_template.md]: Ditto

File: pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+## Add a few positive/negative examples (optional)
```
Comment (queenypingcap) [inline pull_request_template.md]: Ditto

File: pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+- How does this PR work? Need a brief introduction for the changed logic (optional)
+- Describe clearly one logical change and avoid lazy messages (optional)
+- Describe any limitations of the current code (optional)
+
+Please **NOTE** that:
+
+- Do not assume the code is self-evident
+- Do not assume reviewers understand the original issue
+
+## What are the type of the changes? (mandatory)
```
Comment (queenypingcap) [inline pull_request_template.md]: This would be unnecessary because this is the document repo...

File: pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+The currently defined types are listed below, please pick one of the types for this PR by removing the others:
+
+- New feature (non-breaking change which adds functionality)
+- Improvement (non-breaking change which is an improvement to an existing feature)
+- Bug fix (non-breaking change which fixes an issue)
+- Breaking change (fix or feature that would cause existing functionality to not work as expected)
+
+## How has this PR been tested? (mandatory)
```
Comment (queenypingcap) [inline pull_request_template.md]: Ditto

File: pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
+Thank you for working on TiDB documentation! Please read TiDB's [CONTRIBUTING](https://github.com/pingcap/tidb/blob/master/CONTRIBUTING.md) document **BEFORE** filing this PR.
```
Comment (queenypingcap) [inline pull_request_template.md]: Maybe we need a document contributing guide?

**Comment by lilin90** [comment]
@queenypingcap PTAL again

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+-->
+
+## What have you changed? (mandatory)
+
+Please explain **IN DETAIL** what the changes are in this PR and why they are needed:
+
+- Summarize your change (mandatory)
+- How does this PR work? Need a brief introduction for the changed logic (optional)
+- Describe clearly one logical change and avoid lazy messages (optional)
+- Describe any limitations of the current code (optional)
```
Comment (queenypingcap) [inline .github/pull_request_template.md]: This is the doc repo, so why mentioning "code" here?

File: .github/pull_request_template.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+Please **NOTE** that:
+
+- Do not assume the code is self-evident
```
Comment (queenypingcap) [inline .github/pull_request_template.md]: Ditto

**Pull request closed without merging**